### PR TITLE
🎨 create the home landing page #5

### LINKED ISSUE #5: 🎨  Create the home landing page
#  Create the Home Landing page 
---

#### Note: This issue is dependent on #19, #20 and #1, please complete that first.

Quick links: [Summary](#summary) | [Description](#description) | [Technical-Analysis](#Technical-Analysis) | [Acceptance-Criteria](#Acceptance-Criteria)

---

### Summary

As a user, I want a nice hero section on the home page that captures my attention and introduces the page content. Reuse the text-content component created in #20, the image container created in #19 and the header component in #1. I want a good experience on the web and phone.

### Description

Design:
HomeLandingPage

Link to full design:

```
https://xd.adobe.com/view/b98caf27-6549-49e5-8ed8-c5234788df61-ecdf/screen/cc35e19a-24f2-42ce-8803-ee2942839528
```

### Technical Analysis

This feature can be found in libs/pages/home. This feature should be a stand-alone component. The component's name is home-hero-section.

### Acceptance Criteria

1. When rendered the component is pixel perfect compared to the design above.
2. The component is clean and does not have more than 30 lines
3. The component SCSS uses CSS variables for colour, font, ... Connected to the main style.scss
4. The code is isolated to the libs/page/home package, unless a justified reason is specified.
5. The design is responsive for screen, tablet and phone

### PR DESCRIPTION
**Pull request by jbdura**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Reference the issue related to this PR as shown below. Every issue has it's own number you can find the issue numbers [here](https://github.com/italanta/italanta-apps/issues).

Fixes #5

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Screenshot (optional)
Screenshot from 2023-09-07 12-04-11
Screenshot from 2023-09-07 11-53-02

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/features/pages/home/src/lib/components/home-hero-section/home-hero-section.component.scss [hunk 1]
```diff
@@ -0,0 +1,63 @@
+
+.container {
+  background: #F9F8F4 0% 0% no-repeat padding-box;
```
Comment (AnnaL001) [inline libs/features/pages/home/src/lib/components/home-hero-section/home-hero-section.component.scss:3]: For reuse of colors, use CSS variables specified in elements/theming/_variables.scss for specifying colors/background colors

File: libs/features/pages/home/src/lib/components/home-hero-section/home-hero-section.component.scss [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    display: flex;
+    flex-direction:column;
+    align-items: flex-start;
+    padding-left: 40px;
+
+    .text {
+      display: flex;
+      padding-left: 80px;
+      text-align: left;
+      color: #000000;
```
Comment (AnnaL001) [inline libs/features/pages/home/src/lib/components/home-hero-section/home-hero-section.component.scss:18]: Same here

**Comment by jbdura** [comment]
Requested edits in styles complete

File: libs/features/pages/home/src/lib/components/home-hero-section/home-hero-section.component.scss [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    padding-left: 40px;
+
+    .text {
+      display: flex;
+      padding-left: 80px;
+      text-align: left;
+      color: var(--elewa-website-home-hero-color-2);
+      opacity: 1;
+      font-size: x-small;
+      // align-items: flex-start;
```
Comment (AnnaL001) [inline libs/features/pages/home/src/lib/components/home-hero-section/home-hero-section.component.scss:21]: To eliminate dead code, commented out/unused code should be removed

File: libs/features/pages/home/src/lib/components/home-hero-section/home-hero-section.component.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  }
+
+  // Add the imageConfig property
+  imageConfig: ImageConfig = {
+    imageSrc: '',   // Add image URL
+    title: '',      // Add image title
+    visualisation: ImageVisualisation.Stacked, // You can set this to Stacked or any other value as needed
+    maxWidth: '803px',
+    maxHeight: '796px',
+  };
```
Comment (AnnaL001) [inline libs/features/pages/home/src/lib/components/home-hero-section/home-hero-section.component.ts:30]: To keep the component lean and ensure readability and maintainability, the 'buttonData', 'herotexts' and 'imageConfig' variable content can be moved to a Typescript file in the data module

**Review by AnnaL001 — CHANGES_REQUESTED** [review]
Nicely done, just a few adjustments

**Comment by jbdura** [comment]
Requested changes done

File: libs/features/pages/home/src/lib/main/home/home-page.component.html [hunk 5]
```diff
@@ -3,4 +3,22 @@
...
+          <span class="first-line">We’d love to build something</span>
+          <span class="second-line">amazing together!</span>
+        </h4>
+        <div class="mission">
+          <elewa-website-slider-button [buttonData]="buttonData"></elewa-website-slider-button>
+        </div>
+      </div>
+    </div>
+  </elewa-website-footer>
+</div>
```
Comment (LemmyMwaura) [inline libs/features/pages/home/src/lib/main/home/home-page.component.html:24]: This was wrapped by the main page component. you should leave that way. Just add your component where it should be.

File: libs/models/schema/ui/images/src/lib/image-config.ts [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+  maxWidth?: string
+  maxHeight?: string
+}
+
+export enum ImageVisualisation {
+    Pill = 0,
+    Stacked = 1,
+    Window = 2,
+    Card = 3
+  }
```
Comment (LemmyMwaura) [inline libs/models/schema/ui/images/src/lib/image-config.ts:14]: This was deleted. don't add it back

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]

**Comment by jbdura** [comment]
Updated branch with current changes

**Comment by LemmyMwaura** [comment]
Closing this as it's been resolved

**Pull request closed without merging**